### DISCUSSION TITLE
Scaling Compreface

### DISCUSSION THREAD

**Discussion by spb722** [exadel-inc/CompreFace#771] [Q&A]
According to the documentation Scalability can be achieved by increasing the number of  API servers default = 1 and putting an Embedding Servers in a GPU available instance .
1) -  can you tell me where can i make changes for  increasing the number of API servers i couldn't find it in .env file
2) - also will it be possible to add 100000 face collections if were to scale 
3) -  what all changes should i need to make to run embedded server and API servers in separate instance

**Comment by pospielov** [comment]
The easiest way to have a scalable solution is to use Kubernetes. 
https://github.com/exadel-inc/compreface-kubernetes
In this repository, we have configs and instructions on how to run CompreFace with Kubernetes. Just do not forget to update the CompreFace version in config.

> also will it be possible to add 100000 face collections if were to scale

Scaling is needed for increasing the number of requests per second CompreFace can handle. 
Each API node still will store all 100 000 faces in RAM, so you will have to have approximately 20-25 Gb of memory to run the system.
